### PR TITLE
Exception for extracted archives

### PR DESCRIPTION
**Pull request by GPKyte**
Worked from wolever’s commit to integrate with current master.

### CODE REVIEW

File: slackviewer/archive.py [hunk 1]
```diff
@@ -2,14 +2,13 @@
...
+import glob
 
 from slackviewer.message import Message
 
 
 def get_channel_list(path):
-    channels = [d for d in os.listdir(path)
-                if os.path.isdir(os.path.join(path, d))]
-    channels = sorted(channels)
+    channels = [ c["name"] for c in get_channels(path).values() ]
```
Comment (hfaran) [inline slackviewer/archive.py]: list comps don't need the additional space

```diff
-    return [ c["name"] for c in get_channels(path).values() ]
+    return [c["name"] for c in get_channels(path).values()]
```

File: slackviewer/main.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def configure_app(app, archive, debug):
               type=click.INT, help="Host port to serve your content on")
 @click.option("-z", "--archive", type=click.Path(), required=True,
               default=envvar('SEV_ARCHIVE', ''),
-              help="Path to your Slack export archive (.zip file)")
+              help="Path to your Slack export archive (.zip file or directory)”)
```
Comment (hfaran) [inline slackviewer/main.py]: Could you also please update the --help output in the [README in this section](https://github.com/hfaran/slack-export-viewer#installation)

File: slackviewer/archive.py [hunk 3]
```diff
@@ -44,6 +46,10 @@ def SHA1_file(filepath):
 
 
 def extract_archive(filepath):
+    if os.path.isdir(filepath):
+        print("Archive already extracted. Viewing from %s") %(filepath)
+        return(filepath)
```
Comment (hfaran) [inline slackviewer/archive.py]: Statement instead of function for consistency: `return filepath`

File: slackviewer/archive.py [hunk 4]
```diff
@@ -44,6 +46,10 @@ def SHA1_file(filepath):
 
 
 def extract_archive(filepath):
+    if os.path.isdir(filepath):
+        print("Archive already extracted. Viewing from %s") %(filepath)
```
Comment (hfaran) [inline slackviewer/archive.py]:
```python
print("Archive already extracted. Viewing from {}...".format(filepath))
```

**Comment by hfaran** [comment]
Thanks. Just a few nits; then I'll be glad to merge this. :)

**Comment by GPKyte** [comment]
Great thanks!

**Comment by GPKyte** [comment]
Hey, I accidentally pushed a commit that I didn't want to include in this pull request. I'll make the suggested changes and open a new request.

**Pull request closed without merging**